esp/scm/ws_config2.ecm

```diff
@@ -0,0 +1,368 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+ESPstruct statusMsgType
+{
+    string msgLevel;                // message level ("info", "warning", "error", "fatal")
+    string nodeId;                  // node identifier that generated this status message
+    string attribute("");           // name of node's attribute generating the message ("" for the node itself)
+    string msg;                     // the message
+    ESParray<string, parentId> parentIdList;
+};
+
+ESPstruct statusType
+{
+    bool error(false);    // true if a message exsits in status (below) with a message level of error or higher
+    ESParray<ESPstruct statusMsgType, statusMsg> status;
+};
+
+
+ESPresponse PassFailResponse
+{
+    bool error(false);
+    string msg("");
+};
+
+
+ESPresponse PassFailWithStatusResponse
+{
+    bool   error(false);
+    string msg("");
+    ESPstruct statusType status;
+};
+
+
+//
+// Session management (not to be confused with platform session management)
+
+ESPrequest OpenSessionRequest
+{
+    string username;
+    string type("XML");           // type of session
+    string schemaPath("");        // location of configuration schema environmentFiles
+    string masterSchemaFile("");  // name of master schema control file
+    string sourcePath("");        // path to envrionment files
+    string activePath("");        // path to active environment file ("" for none)
+};
+
+
+ESPresponse OpenSessionResponse
+{
+    bool   error(false);
+    string sessionId;
+    string msg("");         // any messages from the open session request
+};
+
+
+ESPrequest CloseSessionRequest
+{
+    string sessionId;
+    bool forceClose(false);    // must be true to close a session that has a modified environmrnt file
+};
+
+
+ESPrequest ListOpenSessionsRequest
+{
+    string dummyParam("");    // not used
+};
+
+
+ESPstruct openSessionInfo
+{
+    string sessionId;
+    string username;
+    string curEnvironmentFile("");
```

Comment:
No, a session can only have one open environment at a time. If you attempt to open another one it will replace the current one, if there are no modifications, or fail if there are unsaved modifications. I could force a close always if you think that would be better.